mllib/src/main/scala/org/apache/spark/mllib/feature/PCA.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.feature
+
+import org.apache.spark.api.java.JavaRDD
+import org.apache.spark.mllib.linalg._
+import org.apache.spark.mllib.linalg.distributed.RowMatrix
+import org.apache.spark.rdd.RDD
+
+/**
+ * Transformer a vector use PCA
+ *
+ * @param k count of principal components
+ */
+
+class PCA(val k: Int) {
+  require(k >= 1)
+
+  /**
+   * Compute a Principal Components from RDD[Vector]
+   *
+   * @param sources source vectors
+   */
+  def fit(sources: RDD[Vector]): PCAModel = {
+    require(k <= sources.first().size)
+
+    val mat = new RowMatrix(sources)
+    val pc = mat.computePrincipalComponents(k) match {
+      case dm: DenseMatrix =>
+        dm
+      /*
+       * Convert Sparse Vector to Dense Vector.
+       *
+       * Following code is for possible compatibility.
+       * RowMatrix.computePrincipalComponents is always returned a Dense Vector
+       */
+      case sm: SparseMatrix =>
+        sm.toDense()
+      case _ =>
+        throw new IllegalArgumentException("Unsupported matrix format. Expected " +
+          s"SparseMatrix or DenseMatrix. Instead got: ${mat.getClass}")
+
+    }
+    new PCAModel(k, pc)
+  }
+
+  /**
+   * Compute a Principal Components from JavaRDD[Vector]
+   *
+   * @param sources source vectors
+   */
+  def fit(k: Int, sources: JavaRDD[Vector]): PCAModel =
+    fit(k, sources.rdd)
+}
+
+/**
+ * This class is wrapper of RowMatrix.computePrincipalComponents and Matrix.multiply
```

Comment:
This is implementation detail but not user-facing doc. `Model fitted by [[PCA]] that can project vectors to a low-dimensional space spanned by principal components.`